### DISCUSSION TITLE
How to make cmp complete trigger automatically when typing?

### DISCUSSION THREAD

**Discussion by rykersun** [hrsh7th/nvim-cmp#1392] [Q&A] [answered]
Hello, I have configuration to trigger `cmp.complete()` when I press `Ctrl + e`,  but is there a way to set it to trigger automatically?

**GIF:**
My current cmp
![ezgif com-gif-maker](https://user-images.githubusercontent.com/86671361/210582351-4da63a63-99b1-426c-8613-a9224a89dd50.gif)

I expect
![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/86671361/210582440-a8577c77-607e-4149-9c14-8871db6692b4.gif)

**Comment by fitrh** [comment, marked as answer]
Make sure your `completion.keyword_length` or `sources[n].keyword_length` is set to 1. How does your cmp config look like?

The following config should open a completion window on every keystroke

```lua
require('cmp').setup({
    completion = {
        keyword_length = 1,
    },
    sources = {
        { name = 'nvim_lsp', keyword_length = 1, },
    },
})
```

**Reply by rykersun** [reply]
Thanks for the reply, but this seems to only fix half of the problem, I still have to press `Ctrl + e` to trigger completion in some cases.

here is my cmp config:

```lua
require('cmp').setup({
  completion = {
    keyword_length = 1,
    max_item_count = 5,
  },
  sources = {
    { name = 'nvim_lsp', keyword_length = 1, max_item_count = 5, },
    { name = 'luasnip', keyword_length = 1, max_item_count = 5, },
    { name = 'buffer', keyword_length = 1, max_item_count = 5, },
  },
})
```

**Reply by fitrh** [reply]
What case and what language server? Some language servers may be slow to return completion items and in this case, cmp can't do anything about it.

You can also set the `completion.keyword_pattern`

```lua
require('cmp').setup({
    completion = {
        keyword_length = 1,
        keyword_pattern = ".*",
    },
    sources = {
        { name = 'nvim_lsp', keyword_length = 1, },
    },
})
```

**Reply by rykersun** [reply]
> What case and what language server? Some language servers may be slow to return completion items and in this case, cmp can't do anything about it.
> 
> You can also set the `completion.keyword_pattern`
> 
> ```lua
> require('cmp').setup({
>     completion = {
>         keyword_length = 1,
>         keyword_pattern = ".*",
>     },
>     sources = {
>         { name = 'nvim_lsp', keyword_length = 1, },
>     },
> })
> ```

This fixed my problem, Thanks a lot!!
